### ISSUE TITLE
Inconsistent behaviour in scenarios with street barrier in leaderboard 1.0.

### ISSUE THREAD

**Issue by PanZWarzywniaka** [carla-simulator/leaderboard#185]
Hi there,

I am currently working on flakiness (non-determinism) in evaluating ADSs in CARLA.

I have been evaluating [TF++](https://github.com/autonomousvision/carla_garage) and discovered that evaluating of same [scenario "2" from evaluation set](https://github.com/carla-simulator/leaderboard/blob/leaderboard-1.0/data/routes_testing.xml) multiple times yields varying results. 

The problem is the collision with the [street barrier](https://carla.readthedocs.io/en/latest/catalogue_props/#street-barrier).
Both the ego car (TF++ agent) and CARLA autopilot vehicles cars just pass through the static object.
[see the video recording](https://www.youtube.com/watch?v=RHkGA8wPomA)

By manually investigating each replay car goes through the barrier every time.
However problem is that collisions are not reported consistently by collision sensor:

<details><summary>Details</summary>
<p>

<table>
<tr><th style="text-align: right;">repetition</th><th style="text-align: right;">N_collisions</th><th style="text-align: right;">DrivingScore</th><th></th></tr>
<tr><td style="text-align: right;">0</td><td style="text-align: right;">3</td><td style="text-align: right;">27.4625</td><td></td></tr>
<tr><td style="text-align: right;">1</td><td style="text-align: right;">2</td><td style="text-align: right;">42.25</td><td></td></tr>
<tr><td style="text-align: right;">2</td><td style="text-align: right;">0</td><td style="text-align: right;">100</td><td></td></tr>
<tr><td style="text-align: right;">3</td><td style="text-align: right;">1</td><td style="text-align: right;">65</td><td></td></tr>
<tr><td style="text-align: right;">4</td><td style="text-align: right;">1</td><td style="text-align: right;">65</td><td></td></tr>
<tr><td style="text-align: right;">5</td><td style="text-align: right;">0</td><td style="text-align: right;">100</td><td></td></tr>
<tr><td style="text-align: right;">6</td><td style="text-align: right;">5</td><td style="text-align: right;">11.6029</td><td></td></tr>
<tr><td style="text-align: right;">7</td><td style="text-align: right;">0</td><td style="text-align: right;">100</td><td></td></tr>
<tr><td style="text-align: right;">8</td><td style="text-align: right;">1</td><td style="text-align: right;">65</td><td></td></tr>
<tr><td style="text-align: right;">9</td><td style="text-align: right;">0</td><td style="text-align: right;">100</td><td></td></tr>
<tr></tr>
</table>

</p>
</details> 

The number of collisions detected with that barrier ranging from `0` to `5` and the driving score ranging from `11.6` to `100`.

To sum up the problem is two fold:

1. CARLA autopilot ignores street barriers as if they don't exist
2. Collision sensor reports diffrent behaviours in the same go-through case.

<details><summary>Software used</summary>
<p>

- OS: Ubuntu 22.04.4 LTS
- CARLA: 0.9.10.1
- Leaderboard 1.0 commit: [2165f5e4a53be8e96c501fa39965800489b1a0a5](https://github.com/carla-simulator/leaderboard/tree/2165f5e4a53be8e96c501fa39965800489b1a0a5)
- ScenarioRunner 1.0 commit: [b9c342eee4ba03cfdec58b184f10f89b7053b65e](https://github.com/carla-simulator/scenario_runner/tree/b9c342eee4ba03cfdec58b184f10f89b7053b65e)

</p>
</details>
